crates/rspack_core/src/code_generation_results.rs

```diff
@@ -133,14 +137,48 @@ impl CodeGenerationResult {
   }
 }
 
+#[derive(Debug, Clone, Copy, Hash, Eq, PartialEq, Ord, PartialOrd, Serialize)]
+pub struct CodeGenResultId(u32);
+
+impl Default for CodeGenResultId {
+  fn default() -> Self {
+    Self(CODE_GEN_RESULT_ID.fetch_add(1, std::sync::atomic::Ordering::Relaxed))
+  }
+}
+
+pub static CODE_GEN_RESULT_ID: Lazy<AtomicU32> = Lazy::new(|| AtomicU32::new(0));
+
 #[derive(Debug, Default)]
 pub struct CodeGenerationResults {
-  // TODO: This should be a map of ModuleIdentifier to CodeGenerationResult
-  pub module_generation_result_map: IdentifierMap<CodeGenerationResult>,
-  map: IdentifierMap<RuntimeSpecMap<ModuleIdentifier>>,
+  pub module_generation_result_map: HashMap<CodeGenResultId, CodeGenerationResult>,
+  map: IdentifierMap<RuntimeSpecMap<CodeGenResultId>>,
```

Comment:
I guess this can refactor to `IdentifierMap<RuntimeSpecMap<CodeGenerationResult>>`, and remove `module_generation_result_map`, let's have a try in the following PR.

Reply:
`CodeGenResultId` is used to avoid replicating, if `usedExports` is opt out, then we need to clone `CodeGenResult` for each related runtimes.https://github.com/web-infra-dev/rspack/pull/4263/files#diff-5eb759d814281903aa0355174051f134df9444a46e9a1ce332cad27eed896215R1035-R1039.